OgreMain/src/OgreRenderQueueSortingGrouping.cpp

```diff
@@ -192,12 +192,10 @@ namespace {
             collection = &mSolidsBasic;
         }
 
-
-        Technique::Passes::const_iterator i;
-        for(i = pTech->getPasses().begin(); i != pTech->getPasses().end(); ++i)
+        for(const auto& p : pTech->getPasses())
```

Comment:
```suggestion
        for(auto* p : pTech->getPasses())
```
(same for the instance below)

Reply:
please also fix the other instances

Reply:
did